eva/readers/document/document_reader.py

```diff
@@ -0,0 +1,33 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pathlib import Path
+from typing import Dict, Iterator
+
+from eva.readers.abstract_reader import AbstractReader
+from eva.readers.document.registry import LOADER_MAPPING
+
+
+class DocumentReader(AbstractReader):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _read(self) -> Iterator[Dict]:
+        ext = Path(self.file_url).suffix
+        assert ext in LOADER_MAPPING, f"File Format {ext} not supported"
+        loader_class, loader_args = LOADER_MAPPING[ext]
+        loader = loader_class(self.file_url, **loader_args)
+        # load entire document as one entry
+        for data in loader.load():
+            yield {"data": data.page_content}
```

Comment:
Is ` data.page_content` always a python `str` regardless of the loader? Type hint here will be useful for understanding.

Reply:
Yes, as per the `langchain` documentation. I'll add a reference.